### PR TITLE
SAK-32614 - Weird content-type prevents attachments to be processed

### PR DESCRIPTION
**Pull request by danielmerino**

### CODE REVIEW

File: mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java [hunk 1]
```diff
@@ -684,6 +684,7 @@ else if (p.isMimeType("text/enriched") && p.getFileName() == null) {
                 name = decodedName;
             }
 
+            type=type.replace("[", "").replace("]", ""); //ContentType throws an Exception when parsing these chars.
```
Comment (buckett) [inline mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java]: The example content type isn't good even if you remove the `[` and `]`. I think a fuller solution would be to have the `new ContentType` in it's own `try{} catch{}` block and falling back to `application/octet-stream`

Comment (danielmerino) [inline mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java]: Ok, Matt, thanks, I will do it as you propose.

**Comment by danielmerino** [comment]
I have done the try catch block solution. Please, could you review it, @buckett ?

File: mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java [hunk 2]
```diff
@@ -684,7 +685,15 @@ else if (p.isMimeType("text/enriched") && p.getFileName() == null) {
                 name = decodedName;
             }
 
-            ContentType cType = new ContentType(type);
+            ContentType cType = null;
+            try
+            {
```
Comment (buckett) [inline mailarchive/mailarchive-subetha/src/java/org/sakaiproject/mailarchive/SakaiMessageHandlerFactory.java]: Brace format doesn't match the rest of this file.

**Comment by danielmerino** [comment]
I don't understand why I must adapt my code format to other's likings. There isn't a community recommendation about brace format AFAIK. So the first committer of each file forces to the next committers to use the same format?

I think it has no sense. Anyway, I have changed it.

**Comment by buckett** [comment]
@danielmerino Yeah, it's a bit of a mess, yes we should have a recommendation and the existing recommendation is to match the file/project. It's just one of those things people haven't tackled.

Thanks for tidying it up.

**Pull request merged**